app/src/main/java/com/devshish/internship/presentation/ui/MainActivity.kt

```diff
@@ -122,6 +120,19 @@ class MainActivity : AppCompatActivity() {
         }
     }
 
+    override fun onBackPressed() {
```

Comment:
https://www.youtube.com/watch?v=Covu3fPA1nQ

Try to avoid this thing with finishing activity on backpress
I think you need to try multiple backstacks, as in video, or clear stack on each of root fragments